### ISSUE TITLE
Set Wave speed or angle?

### ISSUE THREAD

**Issue by coreyog** [lah7/polychromatic#111]
I use a Razer Ornata Chroma at home on Windows and see that it's capable of setting up a Wave lighting effect with the wave being at an arbitrary angle and progressing across the keyboard at a specified pace. At work I use the regular old Ornata on Linux Mint and find that the angle and speed are not adjustable.

I hope someday this can be implemented.

**Comment by lah7** [comment]
That'll come under as a software (custom) effect, something that the daemon people are working on improving AFAIK.

I'm working towards a profile overhaul for **v0.4.0**, of which should hopefully also include a runtime for such animations.

**Comment by ET-Bent** [comment]
Any news on this one?

**Comment by lah7** [comment]
The daemon guys are busy with other things, but improving custom effects is still one of their long-term to-dos.

For Polychromatic, I'm [:snail:] working on refactoring a few areas so it'll be easier to start adding in loads of new/improved stuff, including better profile support and (in this case) incorporate an animation editor. :crossed_fingers: 

No ETA, when it's ready :wink: [(It's not stable for daily builds at the moment either)](https://github.com/lah7/polychromatic/tree/unstable)

**Comment by ET-Bent** [comment]
Cheers! Good to hear. Thank you for the clarification :slightly_smiling_face:

**Comment by Dradeliomecus** [comment]
I am curious about this, is it still to be implemented?

**Comment by lah7** [comment]
Yes, custom effect support is one of the next things for Polychromatic. Then it's a matter of creating a layer that can specify parameters for a "software" wave effect.

**Comment by TCROC** [comment]
Just checking in to see what the status is on this issue. :)

**Comment by lah7** [comment]
Not there yet - just a _little_ more to go before I'm at a point for writing software effects (like custom waves) and attaching them to layers to build effects. :ok_hand: 

As a sneak peak, the editor will look _along_ the lines like this:


![Screenshot_20210102_161915](https://user-images.githubusercontent.com/13032135/103461408-43614700-4d16-11eb-9e2a-09a340757125.png)

_(Imagining the white area is where the device or grid to map will go, and Draw/Picker/Eraser will be renamed accordingly since for **layered effects**, you'll assign keys to a layer)_

**Comment by TCROC** [comment]
Awesome!  Great work!  I'm looking forward to using it! :)  Hope you have been having a great holiday! :)

**Comment by Teskann** [comment]
If you wish to create a customizable wave effect quickly, I created a Python script just for this !
https://github.com/Teskann/razer-waver
